x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -392,6 +389,16 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
     return !!(scalingType === SCALING_TYPES.TOP_HITS && topHitsSplitField);
   }
 
+  async loadIsEditable(): Promise<boolean> {
+    if (!getMapAppConfig().enableDrawingFeature) {
+      return false;
+    }
+    await this.getIndexPattern();
+    const { matchingIndexes } = await getMatchingIndexes(this.indexPattern!.title);
+    // For now we only support 1:1 index-pattern:index matches
+    return matchingIndexes.length === 1;
```

Comment:
matchingIndexes is typed as `matchingIndexes?: string[]` and thus may be undefined if there is a failure, like permissions for something like that. Handle the case where matchingIndexes is undefined.